apps/meteor/server/models/raw/Analytics.ts

```diff
@@ -14,7 +14,11 @@ export class AnalyticsRaw extends BaseRaw<IAnalytic> implements IAnalyticsModel
 	}
 
 	protected modelIndexes(): IndexDescription[] {
-		return [{ key: { date: 1 } }, { key: { 'room._id': 1, 'date': 1 }, unique: true, partialFilterExpression: { type: 'rooms' } }];
+		return [
+			{ key: { date: 1 } },
+			{ key: { 'room._id': 1, 'date': 1 }, sparse: true, name: 'room-id-date-index' },
```

Comment:
While improving the indexes for this PR, I found out we had an index for `room._id` and `date` for the **Analytics** collection that was actually not indexing anything 😢 That's because we applied an incorrect `partialFilterExpression` to it (`type: 'rooms'` won't find any matching document, I guess we intended to have `type: 'messages'`)
Since we committed not to add migrations in minor releases, I decided to fix it and assign a name to the index (so that it is recreated and doesn't throw a duplicated index name error). We can remove the old/incorrect index in the next major

Reply:
We can put this on a separate PR

Reply:
Question: is the index actually not used for any query? If so, I'd say we can leave it like that for now and do a migration to remove it targetting 7.0

And creating the new index alongside it

Reply:
Yes, it's not used anywhere since it has an incorrect `partialFilterExpression` that won't match any document. But our intention was to use it on `saveMessageSent`, which is part of the flow of this engagement dashboard tab, it inserts the data (that's why I thought of including this here since it's also a performance improvement)

Reply:
I've just removed this index improvement from this PR. I'll create another task for this
Please tell me in case you think this should have stayed here